superset-frontend/packages/superset-ui-core/src/time-comparison/getTimeOffset.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ensureIsArray } from '../utils';
+
+const DAY_IN_MS = 24 * 60 * 60 * 1000;
+export const parseDttmToDate = (dttm: string): Date => {
```

Comment:
Isn't `parseDttmToDate` similar to [parse_human_datetime](https://github.com/apache/superset/blob/821c7d7f2c430c4a4294883a66128ba98fd949c5/superset/utils/date_parser.py#L57)? If yes, is there a way we can reuse that so we don't have 2 implementations to maintain?

Reply:
Thank you for pointing this out. parseDttmToDate and parse_human_datetime are very similar. However, parse_human_datetime is in the backend, while parseDttmToDate is also for validating a [DatePicker component](https://github.com/apache/superset/pull/27718/files#diff-76b79c56cfbdda04728e5590bd1927760383d5c44c9d265a828da8804472a271R65-R81) in the frontend. This separation makes unifying them tricky. I'm open to any suggestions you might have.